components/balance/MultipleBalances.vue

```diff
@@ -0,0 +1,172 @@
+      <span class="is-size-7"> {{ $i18n.t('spotlight.total') }}: </span>
+      <span>${{ delimiter(sumUsd) }}</span>
+    </p>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ApiPromise, WsProvider } from '@polkadot/api'
+import { decodeAddress, encodeAddress } from '@polkadot/util-crypto'
+import { CHAINS, ENDPOINT_MAP } from '@kodadot1/static'
+import { NeoSkeleton } from '@kodadot1/brick'
+import { balanceOf } from '@kodadot1/sub-api'
+
+import format from '@/utils/format/balance'
+import { useFiatStore } from '@/stores/fiat'
+
+import type { PalletBalancesAccountData } from '@polkadot/types/lookup'
+import { calculateExactUsdFromToken } from '@/utils/calculation'
+
+interface Token {
+  token: string
+  balance: string
+  usd: number
+}
+
+interface Chain {
+  [key: string]: Token[]
+}
+
+const { accountId } = useAuth()
+
+const status = ref('loading')
+const availableChains = reactive<Chain>({
+  Kusama: [],
+  Statemine: [],
+  Basilisk: [],
+})
+
+const mapToPrefix = {
+  Kusama: 'ksm',
+  Basilisk: 'bsx',
+  Statemine: 'stmn',
+}
+
+function delimiter(amount: string | number) {
+  const formatAmount = typeof amount === 'number' ? amount.toString() : amount
+  const number = parseFloat(formatAmount.replace(/,/g, ''))
+
+  return number.toLocaleString('en-US', {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  })
+}
+
+const fiatStore = useFiatStore()
+const totalUsd = ref([0])
+const sumUsd = computed(() => totalUsd.value.reduce((a, b) => a + b, 0))
+function calculateUsd(amount: string, token = 'KSM') {
+  if (!amount) {
+    return 0
+  }
+
+  const amountToNumber = Number(amount.replace(/\,/g, ''))
+
+  if (token === 'KSM') {
+    return calculateExactUsdFromToken(
+      amountToNumber,
+      Number(fiatStore.getCurrentKSMValue)
+    )
+  }
+
+  return calculateExactUsdFromToken(
+    amountToNumber,
+    Number(fiatStore.getCurrentBSXValue)
+  )
+}
+
+async function getBalance(chainName, token = 'KSM', tokenId = 0) {
+  const currentAddress = accountId.value
+  const prefix = mapToPrefix[chainName]
+  const chain = CHAINS[prefix]
+
+  const publicKey = decodeAddress(currentAddress)
+  const prefixAddress = encodeAddress(publicKey, chain.ss58Format)
+  const wsProvider = new WsProvider(ENDPOINT_MAP[prefix])
+
+  const api = await ApiPromise.create({
+    provider: wsProvider,
+  })
+
+  let currentBalance
+
+  if (tokenId) {
+    const balance = await api.query.tokens.accounts(prefixAddress, tokenId)
+    currentBalance = format(
+      (balance as PalletBalancesAccountData).free,
+      chain.tokenDecimals,
+      false
+    )
+  } else {
+    const balance = await balanceOf(api, prefixAddress)
+    currentBalance = format(balance, chain.tokenDecimals, false)
+  }
+
+  const balance = delimiter(currentBalance)
+  const usd = calculateUsd(balance, token)
+
+  availableChains[chainName].push({
+    token,
+    balance,
+    usd,
+  })
+  totalUsd.value.push(usd)
+
+  await wsProvider.disconnect()
```

Comment:
Try to leverage the identity store imo

Reply:
yes, on the Todo section, I'm planning to store them in the store to improve the skeleton. the question is, should I make a new state or use the existing state `balance`? because the existing `balance` is a kinda different format with multi-chain assets. need feedback btw cc @vikiival @roiLeo @Jarsen136 @daiagi 

in the existing `balance` state, is using `prefix: balance` format
so, it will store like this
```
ksm: 1,
rmrk: 1,
bsx: 1,
stmn: 1,

total: 4
```

in the multi-chain assets, because of `ksm (rmrk2)` and `rmrk` using the same chain (Kusama). I'm thinking to store them like this
```
kusama: 1,
basilisk: 1,
statemime: 1,

total: 3
```

Reply:
Do as you like we will apply new format where it's needed.
Keep in mind we'll need something simple to access in order to check if account has sufficient funds to buy/makeOffer on item (it was pretty easy with urlPrefix)

But I would like to know if we will loose `snek` balance with this changes?